bin/gpspoller.py

```diff
@@ -206,424 +208,732 @@ def _saveGPSStatus(self, gpsstatus):
+                                if gpsdev["activated"]:
+                                    gpsDeviceFound = True
+                                    if "path" in gpsdev:
+                                        gpspath = gpspath + gpsdev["path"] + " "
+
+                    # no device was found, update the status as such
+                    if gpsDeviceFound == False:
+                        self.logger.debug(f"Activation loop {loopcounter=}, {self.gpsstatus=}")
+
+                        # set the gpsstatus dictionary
+                        self.clearGPSStatus()
+                        self.gpsstatus['mode'] = int(report["mode"]) if "mode" in report else 0
+                        self.gpsstatus['error'] = "Waiting on device to become ready"
+
+                        # Save the status of the GPS(s)
+                        self.savePosition()
+
+                else:
+                    self.logger.debug(f"No data was available from GPSD.")
+
+            except Exception as error:
+
+                # Set this to false
+                gpsDeviceFound = False
+                self.logger.debug(f"Unable to connect to GPSD running on {self.gpshost}:  {error}")
+
+                # Break out of this loop
+                break
+
+            # throttle the loop speed
+            if loopcounter < 10:
+                # Wait short amount of time so the loop isn't spinning at warp speed
+                self.stopevent.wait(short_delay)
+
+            else:
+                # Wait a little longer if we've already been looping for a while
+                self.stopevent.wait(long_delay)
+
+
+            # increment the device activation loop counter
+            loopcounter += 1
+
+        #########
+        # End of device activation loop...
+        #########
+
+         
+        # if a device was activated by GPSD, then save the current status 
+        if gpsDeviceFound:
+
+            datetime_record = self.createDateString(gpsd.utc)
+
+            self.clearGPSStatus()
+            self.gpsstatus['utc_time'] = str(datetime_record) if datetime_record else "n/a"
+            self.gpsstatus['mode'] = int(report["mode"]) if "mode" in report else 0
+            self.gpsstatus["status"] = "waiting on device"
+            self.gpsstatus['error'] = "n/a"
+            self.gpsstatus["devicepath"] = str(gpspath)
+
+            self.logger.debug("GPS activated: {}".format(self.gpsstatus))
+
+            # Save the status of the GPS(s)
+            self.savePosition()
+
+
+        # return boolean on if a device was found or not
+        return gpsDeviceFound
+
+
+
+    ##################################################
+    # connect to GPSD
+    ##################################################
+    def connectToGPSD(self, retrydelay: int = 5, retrylimit: int = 0)->gps:
+
+        # initialization
+        loopcounter = 0
+        retrylimit = retrylimit if retrylimit > 0 else 2178310
```

Comment:
I give up - what is the significance of 2178310? It is almost equal to 2 ** 21 but not quite. A Google search for it turns up nothing.  Enlighten me please

Reply:
Aha! Had I read few lines further I would have seen your array of Fibbonacci numbers and figured it out.

Reply:
LOL!  Yeah, just needed a way to skip reporting a failed connection attempt to GPSD.  It will retry connection attempts every 5 seconds (by default), but we don’t want to spit out a log message about that every time through the loop.  So the Fibonacci series seemed like a good way to progressively increase the time between log messages as the retry counts get high.  That way we don’t fill up logs with useless retry failure messages every 5 seconds!!